conjure-go-client/httpclient/failover_test.go

```diff
@@ -50,6 +51,41 @@ func TestFailover503(t *testing.T) {
 	assert.Equal(t, 3, n)
 }
 
+func TestFailover429(t *testing.T) {
+	n := 0
+	handler := http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		n++
+		if n == 3 {
+			rw.WriteHeader(http.StatusOK)
```

Comment:
you can trigger a push to the channel to exit the test very quickly on expected
this way you can also sleep for longer to reduce flakiness

Reply:
can you elaborate? pushing to the channel the timer is using here will make the select statement pass every time, even without backoff. What we are testing here is that backoff occurs

Reply:
I see now.. you're trying to verify that the backoff did happen by guaranteeing that the request took at least a certain amount of time.

Because there aren't strict guarantees for timers you can't actually test this guaranteed this way, so it does seem a bit error prone this way

A foolproof way to test this would be to mock out the retrier and assert that it was used

Reply:
unfortunately we cannot mock the retrier because it is instantiated per `Do` call and is not a field on the struct itself, only its config is

Reply:
If we really cared we could add some kind of supplier function to the struct but I'm not sure it's worth it. How long does the test currently take to run?

Reply:
well there's only 10ms of backoff time and I'm seeing that as the runtime. I can lower this further, it just seemed like it was already pretty negligible. In moving to #81 the middleware can encapsulate the retrier itself which would allow for mocking